### ISSUE TITLE
Interpreter completions ignore `__dir__()` method

### ISSUE THREAD

**Issue by takluyver** [davidhalter/jedi#1027]
Example case:

```python
class CompleteAttrs(object):
    def __getattr__(self, name):
        if name == 'foo':
            return 1
        if name == 'bar':
            return 2
        raise AttributeError(name)
    
    def __dir__(self):
        return ['foo', 'bar'] + object.__dir__(self)

from jedi.api import Interpreter

itp = Interpreter("ca.", [{'ca': CompleteAttrs()}])
for c in itp.completions():
    print(c.name)
```

I expect to see `foo` and `bar` amongst the names found, but they're not there. This is similar to #997, but here we're using an interpreter with an already-initialised object, so it should be easier to use its `__dir__()` method.

Thanks @davidhalter :-)

**Comment by takluyver** [comment]
Thanks!

**Comment by takluyver** [comment]
@davidhalter any chance of a new release soon? The new release of IPython relies on Jedi's completions by default, where previously we had combined Jedi with our own completions, so this has just become very visible to IPython users.

**Comment by davidhalter** [comment]
We're pretty close. I'm actually wrapping up f-string support and another guy is working on some Windows Environment things. Really not that far away. Hopefully two weeks. Maybe a month. Is that OK?

**Comment by takluyver** [comment]
I don't suppose I could prevail on you to do an extra bugfix release in the next few days? I can lend a hand with preparing this if you like. If there's a chance it's still a month off, we might need to consider making a new release of IPython with some workaround. I'd rather spend some time getting the fix out than figuring out a workaround and re-releasing IPython.

**Comment by davidhalter** [comment]
Why don't you release IPython without pinning Jedi to a 0.11.x release? I'm not sure what you're doing now, but since there are not going to API changes that affect you guys in IPython I don't see what's going to stop you from doing that. I cannot guarantee that for 0.13.0, but 0.12.0 should be fine for you guys.

If you then release there shouldn't be any issues and users will automatically be installing jedi 0.12.0 once it's out.

**Comment by takluyver** [comment]
We haven't pinned Jedi. Users can get the new version whenever it's available, no matter what the version number is. I'd just like to make it available sooner, because this issue is affecting our users now.

We had an issue opened for it within about a day of our release, which makes me concerned that there will be a lot of people affected. If a new release of Jedi is still weeks away, I might need to do another release of IPython in the meantime to work around this.

**Comment by davidhalter** [comment]
I would argue that I haven't seen __dir__ being used in the wild a lot. It is probably affecting your users a bit, but not for much longer. I'm really just looking into getting everything right about the environment API. I really don't think it makes sense to release something just about this.

**Comment by davidhalter** [comment]
BTW I have finished f-string support and once #1077 is merged were good to go.

**Comment by davidhalter** [comment]
Released 0.12.0 now. Hope that wasn't too long ;)

**Comment by takluyver** [comment]
Thanks; it probably wasn't too long, but since I was unsure about the release date, I already did a bugfix release of IPython to work around the issue.
